lib/pundit.rb

```diff
@@ -271,6 +265,20 @@ def pundit_user
 
 private
 
+  def pundit_authorize_with_policy(record, query)
```

Comment:
Maybe not the best name? Mostly I'm trying to avoid any possible clashes with custom methods in the user's controller. I would just implement this on a different module/class, but it needs to set `@_pundit_policy_authorized`